### ISSUE TITLE
cancerprof: API Client for extracting data from State Cancer Profiles

### ISSUE THREAD

**Issue by realbp** [ropensci/software-review#637]
Submitting Author Name: Brian Park
Submitting Author Github Handle: @realbp
Repository:  https://github.com/getwilds/cancerprof
Version submitted: 0.1.0
Submission type: Standard
Editor: @ldecicco-USGS
Reviewers: @jromanowska, @ginberg

Archive: TBD
Version accepted: TBD
Language: en

---

-   Paste the full DESCRIPTION file inside a code block below:

```
Package: cancerprof
Title: API Client for State Cancer Profiles
Version: 0.1.0
Authors@R: 
    person("Brian", "Park", , "joon.brianpark@gmail.com", role = c("aut", "cre"),
           comment = c(ORCID = "0009-0008-8274-3057"))
Description: An interface for retrieving data from the NIH NCI State Cancer Profiles API <https://statecancerprofiles.cancer.gov/>. State Cancer Profiles provides information about data topics including demographics, screening and risk factors, cancer incidence, and mortality for US states, counties, and health service areas.
License: MIT + file LICENSE
URL: https://github.com/getwilds/cancerprof, https://getwilds.org/cancerprof/
BugReports: https://github.com/getwilds/cancerprof/issues
Encoding: UTF-8
Roxygen: list(markdown = TRUE)
RoxygenNote: 7.3.1
Imports: 
    cdlTools,
    cli,
    dplyr,
    httr2,
    magrittr,
    rlang,
    stringr,
    utils
Suggests: 
    knitr,
    rmarkdown,
    testthat
Config/testthat/edition: 3
VignetteBuilder: knitr
Depends: 
    R (>= 2.10)

```

## Scope

- Please indicate which category or categories from our [package fit policies](https://ropensci.github.io/dev_guide/policies.html#package-categories) this package falls under: (Please check an appropriate box below. If you are unsure, we suggest you make a pre-submission inquiry.):

	- [x] data retrieval
	- [ ] data extraction
	- [ ] data munging
	- [ ] data deposition
    - [ ] data validation and testing
	- [ ] workflow automation
	- [ ] version control
	- [ ] citation management and bibliometrics
	- [ ] scientific software wrappers
	- [ ] field and lab reproducibility tools
	- [ ] database software bindings
	- [ ] geospatial data
	- [ ] text analysis

- Explain how and why the package falls under these categories (briefly, 1-2 sentences):

cancerprof allows users to retrieve data from [State Cancer Profiles](https://statecancerprofiles.cancer.gov) for programmable analysis. cancerprof makes accessing the undocumented API from State Cancer Profiles intuitive and easy.

-   Who is the target audience and what are scientific applications of this package?

The target audience for cancerprof is anyone who wants to access data from state cancer profiles to conduct programmable analysis without having to navigate the complex nature of its GUI. Specifically, cancer researchers could use cancerprof to conduct reproducable analysis of cancer crossed references with a variety of topics found within the data from state cancer profiles.

-   Are there other R packages that accomplish the same thing? If so, how does yours differ or meet [our criteria for best-in-category](https://ropensci.github.io/dev_guide/policies.html#overlap)?

Currently there are no other softwares or packages that extracts the publicly available data from State Cancer Profiles.

-   (If applicable) Does your package comply with our [guidance around _Ethics, Data Privacy and Human Subjects Research_](https://devguide.ropensci.org/policies.html#ethics-data-privacy-and-human-subjects-research)?

Cancerprof does not breach any data privacy laws and complies with the ethics policies of ropensci.

-   If you made a pre-submission inquiry, please paste the link to the corresponding issue, forum post, or other discussion, or @tag the editor you contacted.

https://github.com/ropensci/software-review/issues/635

-   Explain reasons for any [`pkgcheck` items](https://docs.ropensci.org/pkgcheck/) which your package is unable to pass.

Cancerprof passes all pkgcheck items

## Technical checks

Confirm each of the following by checking the box.

- [x] I have read the [rOpenSci packaging guide](https://devguide.ropensci.org/building.html).
- [x] I have read the [author guide](https://devdevguide.netlify.app/authors-guide.html) and I expect to maintain this package for at least 2 years or to find a replacement.

This package:

- [x] does not violate the Terms of Service of any service it interacts with.
- [x] has a CRAN and OSI accepted license.
- [x] contains a [README with instructions for installing the development version](https://ropensci.github.io/dev_guide/building.html#readme).
- [x] includes [documentation with examples for all functions, created with roxygen2](https://ropensci.github.io/dev_guide/building.html#documentation).
- [x] contains a vignette with examples of its essential functions and uses.
- [x] has a [test suite](https://ropensci.github.io/dev_guide/building.html#testing).
- [x] has [continuous integration](https://ropensci.github.io/dev_guide/ci.html), including reporting of test coverage.

## Publication options

- [x] Do you intend for this package to go on CRAN?
- [ ] Do you intend for this package to go on Bioconductor?

- [ ] Do you wish to submit an Applications Article about your package to [Methods in Ecology and Evolution](http://besjournals.onlinelibrary.wiley.com/hub/journal/10.1111/(ISSN)2041-210X/)? If so:

<details>
<summary>MEE Options</summary>

- [x] The package is novel and will be of interest to the broad readership of the journal.
- [ ] The manuscript describing the package is no longer than 3000 words.
- [ ] You intend to archive the code for the package in a long-term repository which meets the requirements of the journal (see [MEE's Policy on Publishing Code](http://besjournals.onlinelibrary.wiley.com/hub/journal/10.1111/(ISSN)2041-210X/journal-resources/policy-on-publishing-code.html))
- (*Scope: Do consider MEE's [Aims and Scope](http://besjournals.onlinelibrary.wiley.com/hub/journal/10.1111/(ISSN)2041-210X/aims-and-scope/read-full-aims-and-scope.html) for your manuscript. We make no guarantee that your manuscript will be within MEE scope.*)
- (*Although not required, we strongly recommend having a full manuscript prepared when you submit here.*)
- (*Please do not submit your package separately to Methods in Ecology and Evolution*)

</details>

## Code of conduct

- [x] I agree to abide by [rOpenSci's Code of Conduct](https://ropensci.org/code-of-conduct/) during the review process and in maintaining my package should it be accepted.

**Comment by ldecicco-USGS** [comment]
@ropensci-review-bot assign @ldecicco-USGS as editor

**Comment by ldecicco-USGS** [comment]
### Editor checks:

- [x] **Documentation**: The package has sufficient documentation available online (README, pkgdown docs) to allow for an assessment of functionality and scope without installing the package. In particular,
    - [x] Is the case for the package well made?
    - [x] Is the reference index page clear (grouped by topic if necessary)?
    - [x] Are vignettes readable, sufficiently detailed and not just perfunctory?
- [x] **Fit**: The package meets criteria for [fit](https://devguide.ropensci.org/policies.html#package-categories) and [overlap](https://devguide.ropensci.org/policies.html#overlap).
- [x] **Installation instructions:** Are installation instructions clear enough for human users?
- [x] **Tests**: If the package has some interactivity / HTTP / plot production etc. are the tests using [state-of-the-art tooling](https://devguide.ropensci.org/building.html#testing)?
- [x] **Contributing information**: Is the documentation for contribution clear enough e.g. tokens for tests, playgrounds?
- [x] **License:** The package has a CRAN or OSI accepted license.
- [x] **Project management**: Are the issue and PR trackers in a good shape, e.g. are there outstanding bugs, is it clear when feature requests are meant to be tackled?
---

#### Editor comments
There could be more information added to the README, although the bare minimum to meet our criteria is there. 

In the examples I tried, my first thought was it might be nice to convert some of the text. For instance:

```r
x <- demo_income(
  area = "usa",
  areatype = "state",
  income = "median family income",
  race = "all races (includes hispanic)"
)
head(x$Rank)
[1] "52 of 52" "51 of 52" "50 of 52" "49 of 52" "48 of 52" "47 of 52"
```
Seems like c(52, 51, 50, etc) would be a more useful output to an R user. You'd probably want/need another column or _something_ to give the user the " of 52". Not mandatory, could be handy though (maybe a simple function to offer users outside of the function? or a simple example within the examples for how to extract the rank number).

---

**Comment by ldecicco-USGS** [comment]
@ropensci-review-bot seeking reviewers

**Comment by realbp** [comment]
Thank you for the feedback! I will make those changes in the upcoming version of cancerprof. I have added the ropensci badge and created a NEWS.md file. 

What are the next steps in the review process?

**Comment by ldecicco-USGS** [comment]
I'm asking around to find 2 reviewers. Hopefully that shouldn't take too long!

**Comment by ldecicco-USGS** [comment]
@ropensci-review-bot assign @jromanowska as reviewer

**Comment by jromanowska** [comment]
Hi! Just for your information: I'll start with the review soon. There are many free days in May here, in Norway, but I hope I will not need any extension of the review deadline. :crossed_fingers:

**Comment by jromanowska** [comment]
@ldecicco-USGS , I just wanted to notify that {goodpractice} package that is dependency for {pkgcheck} has been archived by CRAN (https://cran.r-project.org/web//packages/goodpractice/index.html) so I couldn't install {pkgcheck} and had to install the GitHub version of {goodpractice} by hand.

**Comment by jromanowska** [comment]
Hi, I'm having problems installing the package:

``` r
pak::pak("getwilds/cancerprof")
#> Error: ! error in pak subprocess
#> Caused by error: 
#> ! Could not solve package dependencies:
#> * getwilds/cancerprof: ! pkgdepends resolution error for getwilds/cancerprof.
#> Caused by error: 
#> ! Bad GitHub credentials, make sure that your GitHub token is valid.
#> Caused by error in `stop(http_error(resp))`:
#> ! Unauthorized (HTTP 401).
devtools::install_github("getwilds/cancerprof")
#> Downloading GitHub repo getwilds/cancerprof@HEAD
#> Installing 3 packages: terra, raster, cdlTools
#> Installing packages into ‘/home/jro049/R/x86_64-pc-linux-gnu-library/4.3’
#> (as ‘lib’ is unspecified)
#> trying URL 'https://cloud.r-project.org/src/contrib/terra_1.7-71.tar.gz'
#> Content type 'application/x-gzip' length 836573 bytes (816 KB)
#> ==================================================
#> downloaded 816 KB
#> 
#> trying URL 'https://cloud.r-project.org/src/contrib/raster_3.6-26.tar.gz'
#> Content type 'application/x-gzip' length 576421 bytes (562 KB)
#> ==================================================
#> downloaded 562 KB
#> 
#> trying URL 'https://cloud.r-project.org/src/contrib/cdlTools_1.13.tar.gz'
#> Content type 'application/x-gzip' length 43089 bytes (42 KB)
#> ==================================================
#> downloaded 42 KB
#> 
#> * installing *source* package ‘terra’ ...
#> ** package ‘terra’ successfully unpacked and MD5 sums checked
#> ** using staged installation
#> configure: CC: gcc
#> configure: CXX: g++ -std=gnu++17
#> checking for gdal-config... no
#> no
#> configure: error: gdal-config not found or not executable.
#> ERROR: configuration failed for package ‘terra’
#> * removing ‘/home/jro049/R/x86_64-pc-linux-gnu-library/4.3/terra’
#> ERROR: dependency ‘terra’ is not available for package ‘raster’
#> * removing ‘/home/jro049/R/x86_64-pc-linux-gnu-library/4.3/raster’
#> ERROR: dependencies ‘raster’, ‘terra’ are not available for package ‘cdlTools’
#> * removing ‘/home/jro049/R/x86_64-pc-linux-gnu-library/4.3/cdlTools’
#> 
#> The downloaded source packages are in
#> 	‘/tmp/RtmpyrNDzF/downloaded_packages’
#> ── R CMD build ─────────────────────────────────────────────────────────#> ──────────────────────────────────────────────────
#> ✔  checking for file ‘/tmp/RtmpyrNDzF/remotes24d8468ec6bf/getwilds-cancerprof-23dbd98/DESCRIPTION’ ...
#> ─  preparing ‘cancerprof’:
#> ✔  checking DESCRIPTION meta-information
#> ─  checking for LF line-endings in source and make files and shell scripts
#> ─  checking for empty or unneeded directories
#> ─  looking to see if a ‘data/datalist’ file should be added
#> ─  building ‘cancerprof_0.1.0.tar.gz’
#>    Warning: invalid uid value replaced by that for user 'nobody'
#>    
#> Installing package into ‘/home/jro049/R/x86_64-pc-linux-gnu-library/4.3’
#> (as ‘lib’ is unspecified)
#> ERROR: dependency ‘cdlTools’ is not available for package ‘cancerprof’
#> * removing ‘/home/jro049/R/x86_64-pc-linux-gnu-library/4.3/cancerprof’
#> Warning messages:
#> 1: In i.p(...) : installation of package ‘terra’ had non-zero exit status
#> 2: In i.p(...) : installation of package ‘raster’ had non-zero exit status
#> 3: In i.p(...) :
#>   installation of package ‘cdlTools’ had non-zero exit status
#> 4: In i.p(...) :
#>   installation of package ‘/tmp/RtmpyrNDzF/file24d821d2b458/cancerprof_0.1.0.tar.gz’ had non-zero exit status
#> 
#> 
```

<sup>Created on 2024-04-30 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

**Comment by ldecicco-USGS** [comment]
Do you get the same error if you install terra and raster independently?

```r
install.packages(c("terra", "raster"))
```

**Comment by jromanowska** [comment]
Today I've tried on another computer (also Linux) and the `pak` command worked - it actually showed me which system libraries were missing :thinking: 
After installing those, I could re-run the `pak` without problems. I will write some comments about this installation process and issues in my review, so that other users may be aware.

**Comment by ldecicco-USGS** [comment]
Hi @realbp! I'm sorry I haven't found a 2nd reviewer yet. I sent a few inquires out and didn't get responses back. I should have sent a 2nd batch out but that task got buried in my to-do list. Apologies for the delay! I've sent a few more requests out and hopefully we can get this review process (re)kicked off!

**Comment by jromanowska** [comment]
Hi, I'm sorry - I need another week to complete the review. May is horrible with lots of free days here in Norway, and a fantastic summer weather this year, leaving too little time for work! :sweat_smile:

**Comment by ldecicco-USGS** [comment]
@ropensci-review-bot assign @ginberg as reviewer

**Comment by jromanowska** [comment]
Thanks for the patience! Great work with {cancerprof}! :raised_hands: 

I'm done with reviewing, please check my comments below.

## Package Review

- **Briefly describe any working relationship you have (had) with the package authors.**    
No working relationship.
- [x] As the reviewer I confirm that there are no [conflicts of interest](https://devguide.ropensci.org/policies.html#coi) for me to review this work (if you are unsure whether you are in conflict, please speak to your editor _before_ starting your review).

### Documentation

The package includes all the following forms of documentation:

- [ ] **A statement of need:** clearly stating problems the software is designed to solve and its target audience in README
  - not really - the authors could expand the description by adding 1-2 sentences;
  - there is no order in vignettes/articles, so it's difficult to navigate the
  help page

- [ ] **Installation instructions:** for the development version of package and any non-standard dependencies in README
  - the required system libraries could be written down (e.g., on Linux, I had
  to install some separately before installing the package);
  - while `pak` is a great package, some people might not use it and try installing
  `cancerprof` via `devtools` which might fail not giving useful information about
  source of error - the authors could add why they recommend installing via `pak`

- [ ] **Vignette(s):** demonstrating major functionality that runs successfully locally
  - I could not launch vignettes locally - it says `no vignettes found` when
  installed
  - I tried reading the vignettes on the webpage - I have some comments to those,
  please check [below](#is-the-documentation-clear-and-sufficient)

- [x] **Function Documentation:** for all exported functions

- [ ] **Examples:** (that run successfully locally) for all exported functions
  - almost all examples are `not run` - it would be more useful if each example
  showed actually that the function works; this would also make the documentation
  better

- [x] **Community guidelines:** including contribution guidelines in the README or CONTRIBUTING, and DESCRIPTION with `URL`, `BugReports` and `Maintainer` (which may be autogenerated via `Authors@R`).

### Functionality

- [x] **Installation:** Installation succeeds as documented.
- [x] **Functionality:** Any functional claims of the software have been confirmed.
- [x] **Performance:** Any performance claims of the software have been confirmed.
- [x] **Automated tests:** Unit tests cover essential functions of the package and a reasonable range of inputs and conditions. All tests pass on the local machine.
- [ ] **Packaging guidelines**: The package conforms to the rOpenSci packaging guidelines.
  - README could be more detailed, as mentioned above
  - webpage is nice (although the vignettes should be changed, see the comments below)
  - when checking with `devtools::check()` I get a warning:
  `Warning: program compiled against libxml 210 using older 209` - I'm not sure
  where it comes from
  - tests were nicely written, using `testthat`, but maybe the `test-dput*R`
  files that are in the `R` directory should be in the `tests`?

Estimated hours spent reviewing: 16

- [x] Should the author(s) deem it appropriate, I agree to be acknowledged as a package reviewer ("rev" role) in the package DESCRIPTION file.

---

## Review Comments

Overall, the package is really useful for batch retrieval of data.
The functions are well documented and designed, tests are well written.
I have some comments that I hope will improve the usability of both, the
package and documentation.

### Are there improvements that could be made to the code style?

1. Avoid long lines, e.g., lines 59-63 in `demo-education.R` could be split
after each logical operator. This will also improve readability of the if
clauses.
    - run `goodpractice::gp()` to get more tips

2. Running the `styler` package showed that only 4 files could be improved -
great work! You don't need to, but you can try that on these files:
    - R/risk-womens-health.R
    - R/risk-smoking.R
    - R/demo-poverty.R
    - R/demo-population.R


### Is there code duplication in the package that should be reduced?

1. When dealing with specific text, like the various levels of smoking, try to
gather these in one place.    
Currently, these various levels of smoking are 
defined in both `risk-smoking.R` and `handle-smoking.R`. This is difficult to
maintain. One way to deal with that is to create an environment that is loaded
when the package is loaded. Then, you can just grab the entire objects from
this environment, without the need for the user to be bothered by these objects.
[Check here](https://r-pkgs.org/code.html#sec-code-onLoad-onAttach) for some
more explanation.

### Are there user interface improvements that could be made?

1. Use the default values of a function to ease function calling
    - if there is only one possible value for a certain argument, implement this
  as a default value (I understand that it might change in the future but
  it could be default without any problem), e.g., `demo_language` _requires_
  the user to write `language = "language isolation"` even if it's the only
  possible value
    - the functions that change their required arguments when one of the
  arguments is chosen are both, difficult to use and difficult to maintain -
  perhaps you can just give a warning that some arguments are omitted?
    - e.g., in `demo_education`, instead of aborting when the user does not provide some
  arguments, one can give a warning and assume _default_ values
    - especially in [`Risk smoking`](https://getwilds.org/cancerprof/articles/risks-vignette.html#risk-smoking)
  there is so many various combinations that it might be difficult to use that
  function when programmatically fetching data - perhaps you could split this
  into smaller, more focused functions?
    - similarly, for some arguments combinations, another argument is required
  to be only of a certain value (e.g., `incidence_cancer` with `cancer = ovary`
  is possible only with `sex = "females"`), thus in such cases, if the user
  provides a wrong argument the function might just return the correct table and
  issue a warning

1. The `area` and `areatype` arguments are not checked anywhere, so when user
puts a wrong value, they get an uninformative error:
```
Error in `process_resp()`:
! Invalid input, please check documentation for valid arguments.
Run `rlang::last_trace()` to see where the error occurred.
```

### Are there performance improvements that could be made?

No issues here.
The performace depends here on the speed of internet connection of the user.

### Is the documentation clear and sufficient?

#### Installation

1. I got error when trying to install the package locally, from the cloned GitHub repo:
`gdal-config not found` - I can't find any mention that any software like
`gdal` is reqiured by `cancerprof`
2. trying to install from the net, using `pak`, as the instructions say, I get the error:

```
pak::pak("getwilds/cancerprof")
Error:                                                           
! error in pak subprocess
Caused by error: 
! Could not solve package dependencies:
* getwilds/cancerprof: ! pkgdepends resolution error for getwilds/cancerprof.
Caused by error: 
! Bad GitHub credentials, make sure that your GitHub token is valid.
Caused by error in `stop(http_error(resp))`:
! Unauthorized (HTTP 401).
Type .Last.error to see the more details.
```
_(I got this resolved afterwards)_

#### Vignettes and function documentation

1. The vignettes are basically a list of examples for each of the functions.
These examples should be (and in most cases are) in the function documentation. Instead,
a vignette should guide a user through a specific use case scenario.
Could the authors create a set of scientific questions that the package helps to
answer? Especially batch retrieval of data is important.
1. I could not find explanation of "FIPS"
1. In "Demo mobility" it says: "The function defaults to `"all races"`,
`"both sexes"`, `"ages 1+"`" - what does it mean? That the data from the source
is available only for these categories?    
Usually, when I see "defaults", I would assume that these are the values
assumed _if_ there is no specific value for an argument. Here, it seems like
there can't be any other value.

1. There is a mistake in `risk_women_health` documentation - wrong names of the
columns of returned data frame.
1. The authors could explain the different categories of arguments more, e.g.,
in `risk_women_health`, the `women_health` argument has three possible values
("pap smear in past 3 years, no hysterectomy, ages 21-65", "mammogram in past 2 years, ages 50-74", "mammogram in past 2 years, ages 40+") and the columns of the returned data frame are not informative
enough to understand what type of data one actually gets

1. Is there a typo in `risk_alcohol` documentation, under `alcohol` argument?
(and does this argument value need to be so long?)

#### Does the documentation use the principle of multiple points of entry i.e. takes into account the fact that any piece of documentation may be the first encounter the user has with the package and/or the tool/data it wraps?

1. The webpage is well built, so that one can get to the home page easily
1. However, there is no common help-page for the package (i.e.,
  `help(package = "cancerprof")` gives
  `URL '/help/library/cancerprof/html/00Index.html' not found`)

### Were functions and arguments named to work together to form a common, logical programming API that is easy to read, and autocomplete?

Yes - good choice of naming. Although the `demo_` functions sounded at first
like "demonstration" to me, but it's logical to abbreviate "demography" as "demo". :)

**Comment by realbp** [comment]
@jromanowska Thank you so much for your hard work at reviewing cancerprof! I will get to work on implementing improvements based on your feedback.

**Comment by ldecicco-USGS** [comment]
@ropensci-review-bot submit review https://github.com/ropensci/software-review/issues/637#issuecomment-2129520979 time 16

**Comment by ldecicco-USGS** [comment]
Thank you @jromanowska for the thorough review!

**Comment by ldecicco-USGS** [comment]
Just a heads up that I'll be on leave from June 17-July 8 and may or may not be able to get to this issue during that time.

**Comment by jromanowska** [comment]
I will be on holidays entire July. I can probably read and comment here, but I won't be able to test anything.

**Comment by ginberg** [comment]
Well done with creating this package!
See below for my comments.

## Package Review

- **Briefly describe any working relationship you have (had) with the package authors.**
- [x] As the reviewer I confirm that there are no [conflicts of interest](https://devguide.ropensci.org/policies.html#coi) for me to review this work (if you are unsure whether you are in conflict, please speak to your editor _before_ starting your review).

#### Documentation

The package includes all the following forms of documentation:

- [ ] **A statement of need:** clearly stating problems the software is designed to solve and its target audience in README
The README is very brief. Please add some explanation about what type of data can be obtained. Also can you mention the target audience? 
- [x] **Installation instructions:** for the development version of package and any non-standard dependencies in README
I had no problems installing the package
- [ ] **Vignette(s):** demonstrating major functionality that runs successfully locally
It's great that there is a vignette about each data topic. What I miss is an overview of State Cancer Profiles and how these vignettes are connected. I think an overview vignette that contains a description about State Cancer Profiles with links to the other vignettes is useful. In this vignette you could also mention which type of user could use a certain vignette.
- [x] **Function Documentation:** for all exported functions
- [ ] **Examples:** (that run successfully locally) for all exported functions
Each exported function has examples but they are mostly inside a dontrun, so they don't actually run. It would be useful if at least some of them can be run locally.
- [x] **Community guidelines:** including contribution guidelines in the README or CONTRIBUTING, and DESCRIPTION with `URL`, `BugReports` and `Maintainer` (which may be autogenerated via `Authors@R`).

#### Functionality

- [x] **Installation:** Installation succeeds as documented.
- [x] **Functionality:** Any functional claims of the software have been confirmed.
- [x] **Performance:** Any performance claims of the software have been confirmed.
- [x] **Automated tests:** Unit tests cover essential functions of the package and a reasonable range of inputs and conditions. All tests pass on the local machine.
- [x] **Packaging guidelines**: The package conforms to the rOpenSci packaging guidelines.

Estimated hours spent reviewing: 6

- [x] Should the author(s) deem it appropriate, I agree to be acknowledged as a package reviewer ("rev" role) in the package DESCRIPTION file.

---

### Review Comments
- Does the code comply with general principles in the Mozilla reviewing guide?
If there is a parameter in a function that can only have 1 value (e.g. 'crowding' in demo_crowding), you don't need to make it a parameter at all. Or set the value as default. Try to make it as simple as possible for the user to use the function. 
- Is there code duplication in the package that should be reduced?
There is some code duplication, especially related to texts. For example the text in handle_alcohol. Better make it a variable so it will be easier to update.
- Is the documentation (installation instructions/vignettes/examples/demos) clear and sufficient? 
the README and vignettes could be improved (see above)
- Other
The package relies heavily on a working internet connection, if it's not available the tests will fail.
```
ℹ Testing cancerprof
✔ | F W  S  OK | Context
✖ | 3        1 | demo-crowding                                      
────────────────────────────────────────────────────────────────────
Error (test-demo-crowding.R:8:3): Output data type is correct
<httr2_failed/rlang_error/error/condition>
Error: Could not resolve host: statecancerprofiles.cancer.gov
Backtrace:
    ▆
 1. ├─cancerprof::demo_crowding(...) at test-demo-crowding.R:8:2
 2. │ └─... %>% req_perform() at cancerprof/R/demo-crowding.R:57:2
 3. └─httr2::req_perform(.)
 4.   └─base::tryCatch(...)
 5.     └─base (local) tryCatchList(expr, classes, parentenv, handlers)
 6.       └─base (local) tryCatchOne(expr, names, parentenv, handlers[[1L]])
 7.         └─value[[3L]](cond)
```
 It would be good to fail gracefully with an informative message if a resource is not available (and not give a error). See [here](https://forum.posit.co/t/internet-resources-should-fail-gracefully/49199) for some ideas. You probably need to fix this anyway when you want the package to be on CRAN.

**Comment by ldecicco-USGS** [comment]
@ropensci-review-bot submit review https://github.com/ropensci/software-review/issues/637#issuecomment-2163610098 time 6

**Comment by ldecicco-USGS** [comment]
Awesome, thanks @ginberg ! 

@realbp go ahead and start responding to the reviews. I'll try to jump in when I can, but will be a little slow for the next 3 weeks.

**Comment by ldecicco-USGS** [comment]
Hi @realbp , checking in to see if you've got any updates to respond to the reviewers.

**Comment by seankross** [comment]
Hi @ldecicco-USGS, I supervised @realbp's work on this package as part of his internship at the Fred Hutch Data Science Lab, which has sadly ended. We loved working with Brian! Would it be possible for me to take reigns of this review? I plan to be the maintainer of this package once we eventually submit it to CRAN.

**Comment by ldecicco-USGS** [comment]
No problem!

**Comment by ldecicco-USGS** [comment]
@seankross checking in to see how things are going. From our end, we're waiting on a response to the reviews.

**Comment by emilyriederer** [comment]
Hi @seankross - are there any updates here? Given the long delay in review, I may temporarily move this review to "holding" status, but we are happy to see the review through to completion if you'll have time to loop back.

**Comment by maurolepore** [comment]
Dear @realbp and @seankross this is to mark the start of my EiC rotation. I'm reviewing all open issues and leaving a short note to myself of what I see.

I see @seankross replaced @realbp in the role of the author. Both reviews are in and we're waiting for the author to respond but it's taking a while so the previous EiC put it on hold.

@seankross could you please let us know if you intend to resume or stay on hold? It's totally OK to stay on hold since we understand priorities can change. But please do touch base to avoid closing the submission for lack of response.

---

https://devguide.ropensci.org/softwarereview_policies.html#policiesreviewprocess

> * If the author hasn’t requested a holding label, but is simply not responding, we should close the issue within one month after the last contact intent.

**Comment by maurolepore** [comment]
Dear @realbp and @seankross I'm wrapping up my EiC rotation and I see no activity since I started. We should probabbly close. I emailed @seankross for confirmation. 

In any case, I hope you're well and thanks a lot for sharing your work with rOpenSci. I look forward to working with you again whenever it's a good time.

**Comment by seankross** [comment]
Hey @maurolepore would you mind re-sending the email you sent me? I can't track it down. Very sorry about the radio silence, I'll follow up here to respond to these previous comments before Wednesday.

**Comment by maurolepore** [comment]
Dear @seankross, this is to mark the start of my new EiC rotation.

A brief summary of what I see:
- Both reviews have been in for over a year.
- You confirmed in May 2025 that you intended to follow up.
- Since then there have been a few EiC check-ins, but no response on this thread.

I'd like to acknowledge your stated intent to continue, and at the same time the amount of time that has passed without an update. For context, our policy says:

> If the author hasn't requested a holding label, but is simply not responding, we should close the issue within one month after the last contact intent.
> — https://devguide.ropensci.org/softwarereview_policies.html#policiesreviewprocess

I'll ping you on Slack and email and look forward to hearing from you within a month.

Thanks!

**Comment by maurolepore** [comment]
Following our guidelines I'll go ahead and close this issue. 

Thanks everyone for your participation! 🙏
